UniConnectApp/app/src/main/java/com/example/uniconnect/dao/IUniversityDAO.kt

```diff
@@ -11,5 +11,5 @@ interface IUniversityDAO {
      * @return List of universities
      */
     @GET("/search")
-    fun getAllUniversities() : Call<ArrayList<University>>
+    fun getAllUniversities() : Call<List<University>>
```

Comment:
Since for other you been using List so just use List instead of arrayList

Reply:
Good.  This is the "D" in SOLID: Dependency Inversion.  Rely on the interface, not the concrete implementation.